crates/paimon/src/error.rs

```diff
@@ -42,4 +42,21 @@ pub enum Error {
         message: String,
         source: opendal::Error,
     },
+    #[snafu(
+        visibility(pub(crate)),
+        display("Paimon hitting unsupported io error {}", message)
+    )]
+    IoUnsupported { message: String },
+    #[snafu(visibility(pub(crate)), display("Failed to parse URL: {:?}", source))]
+    UrlParse { source: url::ParseError },
```

Comment:
I don't think we should expose a URL parse error directly to the user. Perhaps we can use `ConfigInvalid` with an appropriate message instead.

Reply:
I think your suggestion is better. This error should not be exposed to users. Users do not need to be aware of the error reported by a dependent library.